src/backends/torch/torchinputconns.cc

```diff
@@ -905,31 +935,15 @@ namespace dd
           _ids.push_back(_fnames[vecindex] + " #" + std::to_string(tstart)
                          + "_" + std::to_string(seq.size() - 1));
       }
-    for (long int ti = tstart; ti < tstart + _timesteps; ++ti)
-      {
-        std::vector<double> datavec;
-        std::vector<double> labelvec;
 
-        for (unsigned int li = 0; li < label_size; ++li)
-          labelvec.push_back(seq[ti]._v[_label_pos[li]]);
-        for (int di = 0; di < this->_datadim; ++di)
-          if (std::find(_label_pos.begin(), _label_pos.end(), di)
-              == _label_pos.end())
-            datavec.push_back(seq[ti]._v[di]);
+    if (_train)
```

Comment:
`if (_train)` could be removed if we pass the length of the sequence as an argument to `add_data_instance_labels`